src/modules/ekf2/EKF/aid_sources/gnss/gnss_height_control.cpp

```diff
@@ -140,16 +141,36 @@ void Ekf::controlGnssHeightFusion(const gnssSample &gps_sample)
 					bias_est.reset();
 					resetAidSourceStatusZeroInnovation(aid_src);
 
-				} else {
-					ECL_INFO("starting %s height fusion", HGT_SRC_NAME);
-					bias_est.setBias(-_gpos.altitude() + measurement);
 				}
 
-				aid_src.time_last_fuse = _time_delayed_us;
-				bias_est.setFusionActive();
-				_control_status.flags.gps_hgt = true;
+				bool is_gnss_hgt_consistent = true;
+
+				if (_control_status.flags.gnss_hgt_fault) {
+					if (aid_src.innovation_rejected) {
+						_gnss_hgt_hysteresis_time = 0;
+
+					} else if (_gnss_hgt_hysteresis_time == 0) {
+						_gnss_hgt_hysteresis_time = aid_src.timestamp_sample;
```

Comment:
The `isRecent` and `isTimedOut` functions are using `_time_delayed_us`, not the sample timestamp: https://github.com/PX4/PX4-Autopilot/blob/d205d11c3d15b3c8c92b2770966dcf8d77498ab6/src/modules/ekf2/EKF/ekf.h#L985-L988